### ISSUE TITLE
Install issues and missing device

### ISSUE THREAD

**Issue by wirecuttr** [zlangbert/ha-daikinone#24]
Hi, I can't seem to install this integration. When I try trough HACS, 
I get this error message in HAC:
You are running Home Assistant version '2024.2.0', but this repository requires minimum version '2024.2' to be installed.

If I click the Download button, it changes to a spinner, but doesn't seem to end (at least 5 minutes anyway).
I was able to do a manual install by copying over the custom_components\daikinone folder.

My house has 2 thermostats, 2 furnaces and 2 heat pumps. The integration brought in 7 devices, but is missing one furnace:
![image](https://github.com/zlangbert/ha-daikinone/assets/40327874/7ff31ad5-e37b-4ee5-81b4-c5930d19393e)

(There should be an 'Upstairs Furnace')
My only guess is that it didn't get imported because both furnaces are the same model?

Any idea?

**Comment by zlangbert** [comment]
Hi @wirecuttr!

For the install issue, I can't reproduce that behavior for an unknown reason, but I updated the version requirement to be `2024.2.0`. That will hopefully be fixed in the next release.

Regarding your missing furnace, first question I have is does anything show up in the Home Assistant logs from the integration? Second, can you download the diagnostics on the furnace that is showing up and share that here?

The equipment is keyed off of the model and the serial number. I have a similar setup to yours, and for whatever reason, on one of my pieces of equipment the serial reports as all zeros. If both of your furnaces have invalid serial numbers that could be an explanation.

**Comment by wirecuttr** [comment]
I checked my physical thermostats and apparently both my furnaces show the same serial number!
So the combination of model-serial is not unique!
I imagine that's the issue but the diagnostics is here for the one furnace:
[daikinone-4800615ff48fcd78ee54c1348b086675-Main_Basement Furnace-35e2aaa2f40c0697047186b8134930be.json](https://github.com/zlangbert/ha-daikinone/files/14320645/daikinone-4800615ff48fcd78ee54c1348b086675-Main_Basement.Furnace-35e2aaa2f40c0697047186b8134930be.json)

I'm not sure where to find the logs..is that the Logbook? I don't see anything related to the integration there.

BTW thanks for this integration it's got great amounts of data in it. I'm checking out the various sensors and will post some more information.

**Comment by zlangbert** [comment]
Interesting, you would think that is not something Daikin would get wrong haha. I'm not entirely sure what to do about that.

Logs are in Settings -> System -> Logs. They probably are going to say something about duplicate entity ids. At this point we know the issue so the logs aren't necessary.

On the diagnostics, apologies, I told you the wrong thing. You need to download them from the thermostat device.

**Comment by wirecuttr** [comment]
Logs as expected:
``
Logger: homeassistant.components.sensor
Source: helpers/entity_platform.py:636
Integration: Sensor (documentation, issues)
First occurred: 3:51:47 PM (11 occurrences)
Last logged: 3:51:47 PM

Platform daikinone does not generate unique IDs. ID DM97MC0603BNAB-[SERIAL_NUMBER]-Humidification Demand Requested already exists - ignoring sensor.main_basement_furnace_humidification_demand_requested
Platform daikinone does not generate unique IDs. ID DM97MC0603BNAB-[SERIAL_NUMBER]-Power Usage already exists - ignoring sensor.main_basement_furnace_power_usage
Platform daikinone does not generate unique IDs. ID DM97MC0603BNAB-[SERIAL_NUMBER]-Cool Demand Requested already exists - ignoring sensor.main_basement_furnace_cool_demand_requested
Platform daikinone does not generate unique IDs. ID DM97MC0603BNAB-[SERIAL_NUMBER]-Cool Demand Current already exists - ignoring sensor.main_basement_furnace_cool_demand_current
Platform daikinone does not generate unique IDs. ID DM97MC0603BNAB-[SERIAL_NUMBER]-Dehumidification Demand Requested already exists - ignoring sensor.main_basement_furnace_dehumidification_demand_requested
``

**Comment by wirecuttr** [comment]
Oh as well, my heat pumps are DZ17VS, so you could probably add those to your list of equipment as things seem to work.

**Comment by zlangbert** [comment]
I think what I could do it add the thermostat device id to all the entities' unique ids. This way even in a situation like yours they would be unique.

This is a pretty easy change, but the issue is it will essentially break all existing installations because you can't just change an entity's unique id. If you do, HA will mark the original one as removed and create copies with `_2` at the end of the entity id. It seems there is a way to migrate entity unique ids, I will look into the feasibility of that.

**Comment by wirecuttr** [comment]
That would be great (for me) if you could make that change. I think my problem was caused during servicing. I had an issue with one furnace. I believe the technician swapped the control board to the other furnace during testing. They ended up replacing one control board and I guess it ended up with the same serial number as the other one. I'm not clear how the serial number gets entered into the furnace...
So...this is a scenario that may happen to others due to service technicians not paying great attention to serial numbers.

**Comment by wirecuttr** [comment]
So I did try this and it does seem to work.
in the daikinone.py file if I update:  `eid = f"{model}-{serial}` to `eid = f"{payload.id}-{model}-{serial}` then the other furnace does get created. As you mentioned it does orphan all the previously created devices.

**Comment by wirecuttr** [comment]
Thinking about devices. How does this integration deal with multiple locations? I noticed the daikinskyport integration creates only one device which I believe equates to a location based on the [API_info.md](https://github.com/apetrycki/daikinskyport/blob/master/API_info.md):

Once you have an access token, you can connect and retrieve what locations are available to the account:
```
curl --location --request GET "https://api.daikinskyport.com/locations" \
--header "Accept: application/json" \
--header "Authorization: Bearer <access token>"
```

which returns:
```
[{"id":"<UUID of location>","name":"<Location Name>","address":"<street address>","city":"<city>",
"province":"<state/province>","postalCode":"<zip code>",
"country":"<country>","timeZone":"America/New_York","latitude":<lat>,"longitude":<long>,"hasOwner":true}]
```

And retrieve the devices:
```
curl --location --request GET "https://api.daikinskyport.com/devices" \
--header "Accept: application/json" \
--header "Authorization: Bearer <access token>"
```

I think this approach may be more scalable for any folks that have multiple locations?

**Comment by wirecuttr** [comment]
> I think what I could do it add the thermostat device id to all the entities' unique ids. This way even in a situation like yours they would be unique.
> 
> This is a pretty easy change, but the issue is it will essentially break all existing installations because you can't just change an entity's unique id. If you do, HA will mark the original one as removed and create copies with `_2` at the end of the entity id. It seems there is a way to migrate entity unique ids, I will look into the feasibility of that.

What about checking for duplicates and only adding the payload ID if a duplicate is found?

**Comment by wirecuttr** [comment]
conditional checking seems to work and just adds the one device when I update the repository.

To the DaikinOne class, I added:
```
    __eids = []   #list of all eids used to check for duplicates
```

and then modified:

```
            eid = f"{model}-{serial}"
```
to:
```
            eid = f"{model}-{serial}"
            if eid in self.__eids:
                eid = f"{payload.id}-{model}-{serial}"
            self.__eids.append(eid)
```

**Comment by zlangbert** [comment]
@wirecuttr Releasing a new version that should fix this for you. For new integration entries, it will use a new entity unique id format that includes the thermostat id (implemented in #33). So if you don't mind deleting the entry and re-adding, you should be in good shape there.

Regarding locations, I'm completely ignoring them for now. I think regardless of whether there is one or more locations, we get all the devices back when we list them. It would seem pretty unlikely to me that someone would have equipment at different physical locations / address connected to the same HA instance. If you have a different idea please feel free to open another issue about that.
